lib/cylc/config.py

```diff
@@ -1085,12 +1085,14 @@ def configure_queues(self):
         # Then reassign to other queues as requested.
         warnings = []
         requeued = []
-        for queue in queues:
-            if queue == 'default':
+        for key, queue in queues.copy().items():
+            # queues.copy() is essential here to allow items to be removed from
+            # the queues dict.
```

Comment:
That doesn't work on Python 3 either. (Apparently, `dict.keys()` is a view into the `dict` in Python 3, so it would trigger the same run time error.)